zivid_camera/test/test_zivid_camera.cpp

```diff
@@ -460,6 +465,67 @@ TEST_F(CaptureOutputTest, testCapturePointsXYZ)
   }
 }
 
+TEST_F(CaptureOutputTest, testCapturePointsXYZWithROI)
+{
+  auto points_sub = subscribe<sensor_msgs::PointCloud2>(points_xyz_topic_name);
+  dynamic_reconfigure::Client<zivid_camera::SettingsConfig> settings_client("/zivid_camera/settings/");
+  zivid_camera::SettingsConfig configOriginal;
+  ASSERT_TRUE(settings_client.getDefaultConfiguration(configOriginal, dr_get_max_wait_duration));
+  auto config = configOriginal;
+  config.region_of_interest_box_enabled = true;
+  config.region_of_interest_box_point_o_x = 0;
+  config.region_of_interest_box_point_o_y = 0;
+  config.region_of_interest_box_point_o_z = 600;
+  config.region_of_interest_box_point_a_x = 10;
+  config.region_of_interest_box_point_a_y = 0;
+  config.region_of_interest_box_point_a_z = 600;
+  config.region_of_interest_box_point_b_x = 0;
+  config.region_of_interest_box_point_b_y = 10;
+  config.region_of_interest_box_point_b_z = 600;
+  config.region_of_interest_box_extents_min = -1000;
+  config.region_of_interest_box_extents_max = 1000;
+  ASSERT_TRUE(settings_client.setConfiguration(config));
+
+  enableFirst3DAcquisitionAndCapture();
+  const auto& point_cloud = points_sub.lastMessage();
+  ASSERT_TRUE(point_cloud.has_value());
+
+  const auto point_cloud_sdk = captureViaSDK(Zivid::Settings{
+      Zivid::Settings::Acquisitions{ Zivid::Settings::Acquisition{} },
+      Zivid::Settings::RegionOfInterest::Box{
+          Zivid::Settings::RegionOfInterest::Box::Enabled::yes,
+          Zivid::Settings::RegionOfInterest::Box::PointO{ 0, 0, 600 },
+          Zivid::Settings::RegionOfInterest::Box::PointA{ 10, 0, 600 },
+          Zivid::Settings::RegionOfInterest::Box::PointB{ 0, 10, 600 },
```

Comment:
Also, please verify that the ROI here is defined such that it actually filters point. If it fex covered the entire scene then we would not really test anything. If you need a sameple ROI you can search for regionOfInterestBoxDivideHorizontal in SDK.

Reply:
1. Offset all X coordinates by 1 such that all components of a coordinate are unique
2. Without ROI, the number of NaNs reduce to 118k. With ROI, it becomes 2.3M. It is filtering away points.

Reply:
I think you should also use different Z values, fex 590, 600, 610. The test is better that way because currently the test could hide a potential future bug where fex b_z and o_z got mixed up.

Reply:
Is this a blocker or a nice to have?

Reply:
This should be fixed, it should be easy to fix. We should ensure the tests are robust enough to cover potential corner case bugs in the future.

Reply:
Your call :)

Reply:
It does not make a difference to change Z a tiny bit because the extents are pretty large. We need a much smaller box that's more sensitive to Z.